data_scientist_skills/skill_extraction.py

```diff
@@ -0,0 +1,22 @@
+from skills_and_stopwords import all_words_to_remove
+from sklearn.feature_extraction.text import TfidfVectorizer
+import re
+import pandas as pd
+
+def remove_more_stopwords(text):
+    """Removes unnecessary words from text. Use .apply(remove_more_stopwords) on required df column"""
+    text.split(" ")
+    clean_text = re.sub(r'\b(%s)\b' % '|'.join(all_words_to_remove), '', text)
+    clean_text.replace(',', '')
+    return clean_text
+
+def tdidf(X, min_df=0.1, max_df=0.6, results_df=False):
+    """Performs TD-IDF on pd Series. Results_df=True to return table of results. """
+    tfidf = TfidfVectorizer(ngram_range=(1,2), min_df=min_df, max_df=max_df)
+    X_vec = tfidf.fit_transform(X)
+    res = pd.DataFrame(X_vec.toarray(), columns = tfidf.get_feature_names_out())
+    pd.options.display.max_rows = 500
+    words = (res.sum().sort_values(ascending=False))
+    if results_df:
+        return res
+    return words
```

Comment:
So I want to double check I follow the logic here, if `results_df = False` instead of a pd.DataFrame a series is returned with the consolidated values for each word?

A small not that doesn't matter for our data size: you can put the assignment for words after the if statement. Then that statement is only executed if needed.

Reply:
Yep spot on, good point, can see how that would be an issue on big data sets, all good now, will push again